### PR TITLE
Latest BCF PR

### PR DESCRIPTION
**Pull request by TychonautVII**

### CODE REVIEW

File: src/bcf_clean_overpar.cpp [hunk 1]
```diff
@@ -585,22 +912,29 @@ List bcfoverparRcppClean(NumericVector y_, NumericVector z_,
...
     double restemp = 0.0;
     for(size_t k=0;k<n;k++) {
       restemp = y[k]-allfit[k];
-      rss += restemp*restemp;
+      rss += w[k]*restemp*restemp;
     }
     sigma = sqrt((nu*lambda + rss)/gen.chi_square(nu+n));
     pi_con.sigma = sigma/fabs(mscale);
-    pi_mod.sigma = sigma;
+    pi_mod.sigma = sigma; // Is this another copy paste Error?
```
Comment (TychonautVII) [inline src/bcf_clean_overpar.cpp:926]: @jaredsmurray just saw this comment again, flagging it for you!

Comment (jaredsmurray) [reply src/bcf_clean_overpar.cpp:926]: This line was correct in the original code where divisions by bscale analgous to division by mscale in the line above were dealt with in the bdhet function, I need to walk back through the changes here to make sure it's still OK

File: src/funs.cpp [hunk 2]
```diff
@@ -191,48 +193,98 @@ void getpertLU(tree::tree_p pertnode, size_t pertvar, xinfo& xi, int* L, int* U)
...
+		n0=0.0;
+		sv_tmp.resize(nb);
 	}
-}
 
-void allsuffhet(tree& x, xinfo& xi, dinfo& di, double* phi, tree::npv& bnv, std::vector<sinfo>& sv)
+	void operator()(std::size_t begin, std::size_t end){
+		for(size_t i=begin;i<end;i++) {
+			xx = di.x + i*di.p;
+			y=di.y[i]; // @peter I suspect that di.y is the Rj, this trees residual thing
```
Comment (jaredsmurray) [inline src/funs.cpp:245]: comment is correct

File: src/funs.cpp [hunk 3]
```diff
@@ -191,48 +193,98 @@ void getpertLU(tree::tree_p pertnode, size_t pertvar, xinfo& xi, int* L, int* U)
...
-  tree::tree_cp tbn; //the pointer to the bottom node for the current observations
+	tree::tree_cp tbn; //the pointer to the bottom node for the current observations
 	size_t ni;         //the  index into vector of the current bottom node
 	double *xx;        //current x
 	double y;          //current y
 	
 	bnv.clear();
+
+	// This appears to get the bottom nodes from tree x
+	// and save them into the bottom node pointer vector bnv
```
Comment (jaredsmurray) [inline src/funs.cpp:287]: comment is correct

**Review by jaredsmurray — APPROVED** [review]

**Pull request merged**